### DISCUSSION TITLE
On the road to 0.16

### DISCUSSION THREAD

**Discussion by chombourger** [MentorEmbedded/mtda#203] [Announcements]
Now that we have a rough release process documented (see https://github.com/siemens/mtda/blob/master/docs/dev.rst#release-process), we should follow it :)

For 0.16 to be released by the end of the month, we will create a `-tc1` tag today and ask maintainers/contributors to test. We will ideally want people to test NanoPI (R1 and NEO) and beaglebone-black images. The KVM and docker virtual DUTs should also be tested.

As far as frontends are tested, we should test the Command Line Interface, the interactive console and pytest.
Being able to test the web interface (still limited in functionality) would be a plus.

Feedback can be provided here.

**Comment by chombourger** [comment]
Created [v0.16-tc1](https://github.com/siemens/mtda/releases/tag/v0.16-tc1) tag

**Comment by chombourger** [comment]
`git shortlog` for 0.16-tc1:
```
Cedric Hombourger (38):
      ci: move jobs for Ubuntu to ubuntu.yml
      ci: import gpg signing key in ubuntu.yml
      ci: upload source packages for Ubuntu to PPA
      keyboard/hid: make write_report() non-blocking
      console/serial: instantiate Serial object from probe()
      main: probe keyboard from start()
      support/usb: integrate mtda-usb-functions with mtda itself
      console/*: add role to configure()
      console/usbf: use reasonable defaults depending on role
      support/usb: create an extra ACM function as monitor interface
      meta-isar/recipes-bsps: refactor u-boot recipes
      meta-isar/recipes-kernel: add kernel recipe for the beaglebone-black
      meta-isar: add beaglebone-black machine
      kas/debian: add kas file for the beaglebone-black
      main: start with board off and storage on target
      power/docker: make sure we do not pull all images
      www: allow remote access to the console via web browsers
      meta-isar/recipes-python: build HEAD of mtda
      main: serialize sending of messages over the ZeroMQ socket
      mtda-cli: add -c, --config option for user-defined config file
      scripts/test-using-docker: use MTDA_CONFIG environment variable
      main: check for power_controller in _target_off()
      main: make console_lines() return 0 when there is no console
      power/controller: move constants
      tests/target: test target_uptime()
      main: maintain all timeouts in seconds
      main: check for inactive sessions every 10 seconds
      main: fix handling of power_expiry
      main: do not re-arm session timer from target_status()
      main: add config_set_session_timeout()
      main: add config_set_power_timeout()
      pytest: add Target.status()
      tests: new tests for power timeouts
      power/usbrelay: print name of the line that cannot be found
      www: generate per-client session IDs
      www: add a "Power" button
      docs: add rough release process
      ci: do not publish artifacts for -tc or -rc tags

Vijai Kumar K (5):
      docs: remove transparent canvas r1 block diagram
      gnutls28: disable dh_auto_test
      kas/common/base.yml: uprev to ISAR v0.8
      Migrate to IMAGE_FSTYPES
      power/docker: Fix pulling all tags

Gunther Birk (2):
      power/qemu: Qemu configuration for multiple cores
      power/qemu: make initial size of storage devices configurable

bhargav (2):
      debian: fix installing of both docker and kvm packages
      docs: add mtda docker and kvm instructions

sarath P T (1):
      docs: describe how to use static IP addresses in the FAQ
```

**Comment by chombourger** [comment]
Tested KVM setup using v0.16-rc1

**Comment by bhargavdas** [comment]
Tested : 
  web interface with nanopi 
  web interface with docker
  power toggle button on the web interface

**Reply by bhargavdas** [reply]
@chombourger 
both firefox and chrome chromium renders the buttton and text a bit differently

in chrome /chromium
![in-chrome-chromium](https://user-images.githubusercontent.com/44963044/153243536-15305bde-6025-45f1-97f7-f28ef1ba9ff1.png)

in firefox
![in-firefox](https://user-images.githubusercontent.com/44963044/153243609-479a571b-0b94-462f-a5de-df90c8ed7a8b.png)

**Reply by chombourger** [reply]
ok thanks for the feedback. The current Web-UI is more of a PoC than anything. There is definitely some room for improvement (e.g. resize the console window to be exactly 80x25)

**Reply by bhargavdas** [reply]
seem like it doesn't work on ubuntu (install mtda in ubuntu)
can you please validate it once ?

**Reply by chombourger** [reply]
I will have to try in my VM (my primary Linux host is Debian 11)

**Reply by bhargavdas** [reply]
this is strange, I check in 2 ubuntu hosts (not VMs) , the console is not being rendered.
I think we can leave it as a note for this release as
   only supported in debian 
 and debug the ubuntu issue till next release

**Comment by chombourger** [comment]
Pushed a v0.16-tc2 tag

The following changes were merged since -tc1:
```
Richa Bharti (3):
      ci: auto-assign reviewers when PRs are raised
      ci: auto assign reviewer on pull_request_target
      ci: remove auto-assign for assignee

Badrikesh Prusty (1):
      ci: fix no files found in the path for upload

Bhargav Das (1):
      meta-isar/recipes-conf: add "mtda" to /etc/hosts

Jan Kiszka (1):
      meta-isar/recipes-core: Initialize ISAR_RELEASE_CMD
```

Thanks for your testing and contributions

**Reply by chombourger** [reply]
Pushed v0.16-rc1 tag

The following changes were merged since -rc1:

```
Cedric Hombourger (1):
      storage/qemu: create backing file when 'file' is not set
```

We should now collect as many "Tested-by" lines as we possibly could to get v0.16 out the door
